src/database/operations/create/createBaseField.ts

```diff
@@ -0,0 +1,27 @@
+import { db } from '../../db';
+import { NotFoundError } from '../../../errors';
+import type {
+	BaseField,
+	JsonResultSet,
+	WritableBaseField,
+} from '../../../types';
+
+export const createBaseField = async (
+	createValues: WritableBaseField,
+): Promise<BaseField> => {
+	const { label, description, shortCode, dataType } = createValues;
+	const result = await db.sql<JsonResultSet<BaseField>>(
+		'baseFields.insertOne',
+		{
+			label,
+			description,
+			shortCode,
+			dataType,
+		},
+	);
+	const baseField = result.rows[0]?.object;
+	if (baseField === undefined) {
+		throw new NotFoundError('The base field could not be created.');
```

Comment:
This is a testable line / warrants a test.

Reply:
I'm wrong -- I just realized that if insert failed it would throw / never reach this line.  This conditional really only exists for type guard purposes and I think the only way it could happen is if the query file was incorrectly defined.